iroh-store/src/rpc.rs

```diff
@@ -18,107 +16,131 @@ impl iroh_rpc_types::NamedService for Store {
     const NAME: &'static str = "store";
 }
 
-#[async_trait]
-impl RpcStore for Store {
+#[derive(Clone)]
+pub struct RpcStore(Store);
+
+impl RpcStore {
     #[tracing::instrument(skip(self))]
-    async fn version(&self, _: ()) -> Result<VersionResponse> {
+    async fn version(self, _: VersionRequest) -> VersionResponse {
```

Comment:
just remove the argument?

Reply:
This is the line that would not compile if you were to change this:
```rust
...
Version(req) => s.rpc(req, chan, store, RpcStore::version).await,
...
```